Neos.TimeableNodeVisibility/Classes/Command/TimeableNodeVisibilityCommandController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Neos\TimeableNodeVisibility\Command;
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Cli\CommandController;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\TimeableNodeVisibility\Service\TimeableNodeVisibilityService;
+use Neos\TimeableNodeVisibility\Domain\HandlingResult;
+
+#[Flow\Scope('singleton')]
+class TimeableNodeVisibilityCommandController extends CommandController
+{
+    #[Flow\Inject]
+    protected TimeableNodeVisibilityService $timeableNodeVisibilityService;
+
+    public function executeCommand(string $contentRepository = 'default', bool $quiet = false): void
+    {
+        $contentRepositoryId = ContentRepositoryId::fromString($contentRepository);
+        $workspaceName = WorkspaceName::fromString('live');
+
+        $handlingResult = $this->timeableNodeVisibilityService->handleExceededNodeDates(
+            $contentRepositoryId,
+            $workspaceName
+        );
+
+        if (!$quiet) {
+            $this->output->outputLine(sprintf('Enabled %d nodes with exceeded timed dates.', $handlingResult->countByResult(HandlingResult::RESULT_ENABLED)));
+            foreach ($handlingResult->getByResult(HandlingResult::RESULT_ENABLED) as $result) {
+                $this->output->outputLine(sprintf(
+                        '- NodeAggregateId: %s, DimensionSpacePoint: %s, Label: %s',
+                        $result->node->nodeAggregateId->value,
+                        join(',', $result->node->originDimensionSpacePoint->coordinates),
+                        $result->node->getLabel()
+                    )
+                );
+            }
+
+            $this->output->outputLine(sprintf('Disabled %d nodes with exceeded timed dates.', $handlingResult->countByResult(HandlingResult::RESULT_DISABLED)));
+            foreach ($handlingResult->getByResult(HandlingResult::RESULT_DISABLED) as $result) {
+                $this->output->outputLine(sprintf(
+                        '- NodeAggregateId: %s, DimensionSpacePoint: %s, Label: %s',
+                        $result->node->nodeAggregateId->value,
+                        join(',', $result->node->originDimensionSpacePoint->coordinates),
+                        $result->node->getLabel()
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Run the execute command as daemon.
+     *
+     * @param string $contentRepository The content repository identifier. (Default: 'default')
+     * @param int $ttl The time to live for the daemon in seconds. Set to '0' for infinite. (Default: '900')
+     * @param int $interval Interval in seconds, when the command has to get executed. (Default: '60')
+     * @param bool $quiet Set to false if you need a more verbose output. (Default: 'true')
+     */
+    public function runDaemonCommand(string $contentRepository = 'default', int $ttl = 900, int $interval = 60, bool $quiet = true): void
```

Comment:
is php "stable" enough for this task? I just know that other task runners like flowpack/prunner are written in go or use shell scripts for these daemon jobs. There would be of course a performance penalty of rebooting flow.

But we could of course offer both. This `runDaemonCommand` and a script `Scripts` ;)

_disclaimer: i dont have thaaaat much experience with those long running tasks this is just my mustard to the thing_

Reply:
It's kind of an offer... you are always allowed to implement your own. I thought it might be helpful for smaller projects.